src/components/sets/CellSetsManagerSubscriber.js

```diff
@@ -1,87 +1,192 @@
+    PubSub.publish(CELLS_COLOR, cellColors);
+    PubSub.publish(CELL_SETS_VIEW, new Set(cellIds));
+    PubSub.publish(CELL_SETS_CHANGE, tree);
+  }, [tree]);
 
-  cellSetsSubscriber(msg, cellSets) {
-    this.setState({ cellSets });
-  }
+  // Callback functions
+  const onCheckNodes = useCallback((checkedKeys) => {
+    dispatch({ type: ACTION.CHECK_NODES, checkedKeys });
+  }, []);
 
-  cellsSelectionSubscriber(msg, cellIds) {
-    const { cellSets } = this.state;
-    cellSets.setCurrentSet(cellIds, true);
-  }
+  const onCheckNode = useCallback((targetKey) => {
+    dispatch({ type: ACTION.CHECK_NODE, targetKey });
+  }, []);
 
-  render() {
-    const { cellSets, gridResizeEvent } = this.state;
-    const { datasetId, removeGridComponent } = this.props;
-    return (
-      <TitleInfo
-        title="Cell Sets"
-        isScroll
-        removeGridComponent={removeGridComponent}
-      >
-        <SetsManager
-          setsTree={cellSets}
-          datasetId={datasetId}
-          setsType={setsType}
-          onError={err => PubSub.publish(STATUS_WARN, err)}
-          gridResizeEvent={gridResizeEvent}
-        />
-      </TitleInfo>
-    );
-  }
+  const onExpandNode = useCallback((expandedKeys, targetKey, expanded) => {
+    dispatch({
+      type: ACTION.EXPAND_NODE, expandedKeys, targetKey, expanded,
+    });
+  }, []);
+
+  const onDropNode = useCallback((dropKey, dragKey, dropPosition, dropToGap) => {
+    dispatch({
+      type: ACTION.DROP_NODE, dropKey, dragKey, dropPosition, dropToGap,
+    });
+  }, []);
+
+  const onCheckLevel = useCallback((levelZeroKey, levelIndex) => {
+    dispatch({ type: ACTION.CHECK_LEVEL, levelZeroKey, levelIndex });
+  }, []);
+
+  const onNodeSetColor = useCallback((targetKey, color) => {
+    dispatch({ type: ACTION.SET_NODE_COLOR, targetKey, color });
+  }, []);
+
+  const onNodeSetName = useCallback((targetKey, name, stopEditing) => {
+    dispatch({
+      type: ACTION.SET_NODE_NAME, targetKey, name, stopEditing,
+    });
+  }, []);
+
+  const onNodeSetIsEditing = useCallback((targetKey, value) => {
+    dispatch({
+      type: ACTION.SET_NODE_IS_EDITING, targetKey, value,
+    });
+  }, []);
+
+  const onNodeRemove = useCallback((targetKey) => {
+    dispatch({ type: ACTION.REMOVE_NODE, targetKey });
+  }, []);
+
+  const onNodeView = useCallback((targetKey) => {
+    dispatch({ type: ACTION.VIEW_NODE, targetKey });
+  }, []);
+
+  const onCreateLevelZeroNode = useCallback(() => {
+    dispatch({ type: ACTION.CREATE_LEVEL_ZERO_NODE });
+  }, []);
+
+  const onUnion = useCallback(() => {
+    dispatch({ type: ACTION.UNION_CHECKED });
+  }, []);
+
+  const onIntersection = useCallback(() => {
+    dispatch({ type: ACTION.INTERSECTION_CHECKED });
+  }, []);
+
+  const onComplement = useCallback(() => {
+    dispatch({ type: ACTION.COMPLEMENT_CHECKED });
+  }, []);
+
+  const onView = useCallback(() => {
+    dispatch({ type: ACTION.VIEW_CHECKED });
+  }, []);
```

Comment:
Is there a substantial benefit to using memoizing all these inline functions? I could be wrong, but I was under the impression `useCallback`/`useMemo` should generally be used when there is some heavy computation or to preserve some type of ref equality. https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
Thanks for sharing that blog post, it is very helpful! I was going overboard with the useCallbacks